app/javascript/src/components/editor/ExpandableCodeblock.svelte

```diff
@@ -0,0 +1,76 @@
+<script>
+  import { tick } from "svelte"
+  import { reset as microlight } from "../../microlight"
+  import Expand from "../icon/Expand.svelte"
+
+  export let fullContentLines
+  export let snippetLineCount = 3
+  export let snippetHighlightedLineIndex = 0
+
+  let expanded = false
+  let snippetStartLineIdx
+  let snippetLines = []
+  $: {
+    snippetLines = []
+
+    snippetStartLineIdx = snippetHighlightedLineIndex - Math.max(0, Math.floor(snippetLineCount / 2))
+    for (let i = snippetStartLineIdx; i < snippetStartLineIdx + snippetLineCount; i++) {
+      snippetLines.push(fullContentLines[i])
+    }
+  }
```

Comment:
I personally find these reactive statements can get hard to read and it's nicer to separate it out into a function

```suggestion
  
  $: snippetLines = getLines(snippetLineCount)
  
  function getLines() {
    const lines = []

    snippetStartLineIdx = snippetHighlightedLineIndex - Math.max(0, Math.floor(snippetLineCount / 2))
    for (let i = snippetStartLineIdx; i < snippetStartLineIdx + snippetLineCount; i++) {
      linesines.push(fullContentLines[i])
    }
    
    return lines
  }
```